app/controllers/glossary_terms_controller.rb

```diff
@@ -80,6 +75,47 @@ def destroy
 
   private
 
+  # --------- index private methods
+
+  # should index be filtered?
+  def filter_index?
+    params[:q] || params[:by]
+  end
+
+  def index_filtered
+    query = find_or_create_query(:GlossaryTerm, by: params[:by])
+    show_selected_articles(
+      query,
+      id: params[:id].to_s,
+      always_index: true
+    )
+  end
+
+  def index_full
+    query = create_query(
+      :GlossaryTerm,
+      :all,
+      by: :name
+    )
+    show_selected_glossary_terms(query)
+  end
+
+  # Show selected list of glossary_terms.
+  def show_selected_glossary_terms(query, args = {})
+    includes = @user ? { thumb_image: :image_votes } : :thumb_image
+    args = {
+      action: :index,
+      letters: "glossary_terms.name",
+      num_per_page: 50,
+      include: includes
+    }.merge(args)
+    @links ||= []
+
+    show_index_of_objects(query, args)
+  end
+
```

Comment:
@pellaea now that i'm adding something with `show_index_of_objects` for the first time, gotta say it's been pretty smooth!